### ISSUE TITLE
PACKAGE_BUILDING is not defined when doing poud jail -b

### ISSUE THREAD

**Issue by arrowd** [freebsd/poudriere#1196]
`src/share/poudriere/jail.sh` contains following code in the `install_from_src()` function:

```
	if [ ${BUILD} -eq 0 ]; then
		setup_build_env
		installworld
	else
		buildworld
		installworld
		if [ ${BUILD_PKGBASE} -eq 1 ]; then
			build_pkgbase
		fi
	fi
```

Here the `BUILD` var is controlled by the `-b` flag. Since `setup_build_env()` is called only when we're not passing `-b` it results in the `PACKAGE_BUILDING=yes` variable being missing. This in turn leads to the following block of `share/mk/src.sys.mk` being omitted:

```
.if defined(PACKAGE_BUILDING)
CFLAGS+=	-fmacro-prefix-map=${SRCTOP}=/usr/src -fdebug-prefix-map=${SRCTOP}=/usr/src
.endif
```

Which finally leads to wrong debugging info paths written into resulting binaries.

I'm not 100% sure this is a bug on the Poudriere side, but it feels like it is.

**Comment by bdrewery** [comment]
I'm not following your hints from the code. Poudriere only sets `PACKAGE_BUILDING` in `jail_start()` which is not done for `jail -c -b` or `jail -u -b`. Further we aren't building packages so why should that be set? Even with `-B` for pkgbase, `PACKAGE_BUILDING` is not set. Perhaps it should be, but it would only make sense with `-B`.

**Comment by arrowd** [comment]
Then the bug is on the freebsd-src side. I'll look into `git blame` to figure out why that line is wrapped into `.if defined(PACKAGE_BUILDING)`

**Comment by arrowd** [comment]
The `.if defined(PACKAGE_BUILDING)` change in src tracks back to  https://reviews.freebsd.org/D42569

I'm now confused how this condition could ever be true if you say that `-b` should not set it.

My initial issue was that freebsd-src build with `poudriere jail -b` had wrong debuginfo paths baked in. Removing the `.if defined(PACKAGE_BUILDING)` condition or making poudriere define this var fixes the issue for me. Not sure which one is a correct one.

 @bapt maybe you have thoughts?

**Comment by bdrewery** [comment]
Do you have `PACKAGE_BUILDING` in your poudriere.conf or make.conf? It might get into env that way too.

**Comment by arrowd** [comment]
No, it is the other way around - I want `PACKAGE_BUILDING` to be set so that that condition will fire.

**Comment by bdrewery** [comment]
Something about this confused me but I understand now. We should set `PACKAGE_BUILDING` when building world too.
